app/services/OphanService.scala

```diff
@@ -0,0 +1,183 @@
+package services
+
+import java.net.URLEncoder
+
+import abtests.Allocation
+import enumeratum.{EnumEntry, Enum}
+import models.{ContributorRow, ContributionMetaData, PaymentProvider}
+import okhttp3.{HttpUrl, Request}
+import play.api.{Mode, Environment}
+import play.api.libs.json.{Json}
+import com.gu.okhttp.RequestRunners._
+import services.Ophan.{OphanError, OphanSuccess, OphanResponse}
+import scala.concurrent.{ExecutionContext, Future}
+import cats.implicits._
+
+object Ophan {
+
+  sealed trait OphanResponse
+
+  case class OphanSuccess(response: String) extends OphanResponse
+
+  case class OphanError(`type`: String, message: String, code: String = "", decline_code: String = "") extends Throwable with OphanResponse {
+    override def getMessage: String = s"message: $message; type: ${`type`}; code: $code; decline_code: $decline_code"
+  }
+
+}
+
+
+sealed trait PaymentFrequency extends EnumEntry {
+  val stringValue: String
+}
+
+object PaymentFrequency extends Enum[PaymentFrequency] {
+
+  val values = findValues
+
+  case object OneOff extends PaymentFrequency {
+    val stringValue = "ONE_OFF"
+  }
+
+  case object Monthly extends PaymentFrequency {
+    val stringValue = "MONTHLY"
+  }
+
+  case object Annually extends PaymentFrequency {
+    val stringValue = "ANNUALLY"
+  }
+}
+
+trait Product {
+  val stringValue = "CONTRIBUTION"
+}
+
+case object Contribution extends Product
+
+case class TestData(variantName: String, complete: Boolean, campaignCodes: Option[Set[String]])
+
+case class OphanAcquisitionEvent(
+   viewId: String,
+   browserId: String,
+   product: Product,
+   paymentFrequency: PaymentFrequency,
+   currency: String,
+   amount: Double,
+   visitId: Option[String],
+   amountInGBP: Option[Double],
+   paymentProvider: Option[PaymentProvider],
+   campaignCode: Option[Set[String]],
+   abTests: Set[Allocation],
+   countryCode: Option[String],
+   referrerPageViewId: Option[String],
+   referrerUrl: Option[String]
+) {
+  def toParams: Seq[(String, String)] = {
+    Seq(
+      "viewId" -> this.viewId,
+      "browserId" -> this.browserId,
+      "product" -> this.product.stringValue,
+      "currency" -> this.currency,
+      "paymentFrequency" -> this.paymentFrequency.stringValue,
+      "amount" -> this.amount.toString
+    ) ++
+      this.visitId.map(visitId => "visitId" -> this.visitId.toString) ++
```

Comment:
Since there are so mainly optional values, this might be clearer as:
```
List(
  "visitId" -> visitId.map(_.toString),
  "amountInGBP" -> amountInGBP.map(_.toString),
  ...
).collect { case (k, Some(v)) => k -> v }
```